plugin/pkg/admission/namespace/lifecycle/admission.go

```diff
@@ -45,11 +46,30 @@ func init() {
 type lifecycle struct {
 	*admission.Handler
 	client             clientset.Interface
-	store              cache.Store
 	immortalNamespaces sets.String
+	namespaceInformer  framework.SharedIndexInformer
+	// forceLiveLookupCache holds a list of entries for namespaces that we have a strong reason to believe are stale in our local cache.
+	// if a namespace is in this cache, then we will ignore our local state and always fetch latest from api server.
+	forceLiveLookupCache *lru.Cache
```

Comment:
it takes two calls to delete a namespace.  delete 1 -> active to terminating, delete 2 -> gone from store.  my point being that after the first delete, its not gone, just terminating.